File: solutions/design/380-randomized.py
I'll help you implement a Randomized Set in Python that meets the requirements. Here's the complete implementation:

```python
"""
Randomized Set Implementation in Python
This implementation provides O(1) average time complexity for insert, remove, and getRandom operations.
"""

import random

class RandomizedSet:
    def __init__(self):
        """
        Initialize your data structure here.
        Using a dictionary for O(1) lookups and a list for O(1) random access
        """
        self.value_to_index = {}  # Dictionary to store value -> index mapping
        self.values = []          # List to store actual values
        
    def insert(self, val: int) -> bool:
        """
        Inserts a value to the set. Returns true if the set did not already contain the specified element.
        
        Args:
            val: The value to insert
            
        Returns:
            bool: True if the value was inserted, False if it already existed
        """
        if val in self.value_to_index:
            return False
        
        # Add value to the end of the list and store its index
        self.values.append(val)
        self.value_to_index[val] = len(self.values) - 1
        return True
        
    def remove(self, val: int) -> bool:
        """
        Removes a value from the set. Returns true if the set contained the specified element.
        
        Args:
            val: The value to remove
            
        Returns:
            bool: True if the value was removed, False if it didn't exist
        """
        if val not in self.value_to_index:
            return False
        
        # Get index of the value to remove
        index = self.value_to_index[val]
        last_val = self.values[-1]
        
        # Move the last element to the position of the element to remove
        self.values[index] = last_val
        self.value_to_index[last_val] = index
        
        # Remove the last element
        self.values.pop()
        del self.value_to_index[val]
        
        return True
        
    def getRandom(self) -> int:
        """
        Get a random element from the set.
        
        Returns:
            int: A random element from the set
        """
        return random.choice(self.values)

def main():
    """
    Example usage of the RandomizedSet class
    """
    randomized_set = RandomizedSet()
    
    # Test insertions
    print("Inserting 1:", randomized_set.insert(1))  # Returns True
    print("Inserting 2:", randomized_set.insert(2))  # Returns True
    print("Inserting 2 again:", randomized_set.insert(2))  # Returns False
    
    # Test random selection
    print("Random element:", randomized_set.getRandom())  # Returns 1 or 2 randomly
    
    # Test removal
    print("Removing 1:", randomized_set.remove(1))  # Returns True
    print("Removing 3:", randomized_set.remove(3))  # Returns False
    
    # Test final state
    print("Final random element:", randomized_set.getRandom())  # Returns 2

if __name__ == "__main__":
    main()
```

This implementation provides:

1. A `RandomizedSet` class that supports three main operations:
   - `insert(val)`: Inserts a value if it's not present
   - `remove(val)`: Removes a value if it exists
   - `getRandom()`: Returns a random element from the set

2. The implementation uses:
   - A dictionary (`value_to_index`) for O(1) lookups
   - A list (`values`) for O(1) random access
   - Python's built-in `random.choice()` for random selection

3. Key features:
   - All operations (insert, remove, getRandom) have O(1) average time complexity
   - Proper handling of edge cases
   - Clear documentation and comments
   - Example usage in the main function

4. The implementation follows Python best practices:
   - Clear variable names
   - Proper type hints
   - Docstrings for classes and methods
   - Clean code structure

The code is complete and ready to run, with example usage provided in the `main()` function. It handles edge cases appropriately and provides efficient operations for all required functionality.

File: solutions/design/alternatives/380-randomized.js
/**
 * 380. Randomized
 * Medium
 *
 * RandomizedSet Class Implementation This class implements a data structure that supports: - Inserting elements (no duplicates allowed) - Removing elements - Getting a random element All operations should work in O(1) average time complexity
 *
 * <details>
 * <summary><b>🔍 SOLUTION EXPLANATION</b></summary>
 *
 * ### INTUITION:
 * The key insight for solving Randomized is to understand the core problem pattern
 * and apply the most efficient algorithmic approach.
 *
 * ### APPROACH:
 * 1. Analyze the problem requirements
 * 2. Choose the optimal data structure
 * 3. Implement the solution step by step
 * 4. Handle edge cases appropriately
 *
 * ### WHY THIS WORKS:
 * This approach works because it leverages the fundamental properties of the problem
 * to achieve an efficient solution.
 *
 * ### EXAMPLE WALKTHROUGH:
 * For a typical input, the algorithm processes the data systematically
 * to produce the expected output.
 *
 * </details>
 */

/**
 * RandomizedSet Class Implementation
 * 
 * This class implements a data structure that supports:
 * - Inserting elements (no duplicates allowed)
 * - Removing elements
 * - Getting a random element
 * All operations should work in O(1) average time complexity
 */
class RandomizedSet {
    /**
     * Initialize your data structure here.
     * Using both a Map and Array for O(1) operations
     */
    constructor() {
        // Map to store value -> index mapping for O(1) lookup
        this.valueMap = new Map();
        // Array to store actual values for O(1) random access
        this.values = [];
    }
    
    /**
     * Inserts a value to the set.
     * Returns true if the set did not already contain the specified element.
     * @param {number} val
     * @return {boolean}
     */
    insert(val) {
        // Check if value already exists
        if (this.valueMap.has(val)) {
            return false;
        }
        
        // Add value to array and store its index in map
        this.valueMap.set(val, this.values.length);
        this.values.push(val);
        return true;
    }
    
    /**
     * Removes a value from the set.
     * Returns true if the set contained the specified element.
     * @param {number} val
     * @return {boolean}
     */
    remove(val) {
        // Check if value exists
        if (!this.valueMap.has(val)) {
            return false;
        }
        
        // Get index of value to remove
        const index = this.valueMap.get(val);
        const lastValue = this.values[this.values.length - 1];
        
        // Move last element to the position of element to remove
        this.values[index] = lastValue;
        this.valueMap.set(lastValue, index);
        
        // Remove the value
        this.values.pop();
        this.valueMap.delete(val);
        
        return true;
    }
    
    /**
     * Get a random element from the set.
     * @return {number}
     */
    getRandom() {
        // Generate random index and return corresponding value
        const randomIndex = Math.floor(Math.random() * this.values.length);
        return this.values[randomIndex];
    }
}

/**
 * Example usage:
 * const obj = new RandomizedSet()
 * const param_1 = obj.insert(1)
 * const param_2 = obj.remove(2)
 * const param_3 = obj.getRandom()
 */

// Export the class for use in other modules
module.exports = RandomizedSet;

// Test cases
if (require.main === module) {
    // Create new instance
    const randomizedSet = new RandomizedSet();
    
    // Test insertions
    console.log(randomizedSet.insert(1)); // true
    console.log(randomizedSet.insert(2)); // true
    console.log(randomizedSet.insert(1)); // false (already exists)
    
    // Test random element
    console.log(randomizedSet.getRandom()); // 1 or 2 randomly
    
    // Test removal
    console.log(randomizedSet.remove(1)); // true
    console.log(randomizedSet.remove(3)); // false (doesn't exist)
    
    // Test final state
    console.log(randomizedSet.getRandom()); // 2 (only remaining element)
}